dsc_lib/src/configure/mod.rs

```diff
@@ -152,6 +154,72 @@ impl Configurator {
         Ok(result)
     }
 
+    fn find_duplicate_resource_types(config: &Configuration) -> Vec<String>
+    {
+        let mut map: HashMap<&String, i32> = HashMap::new();
+        let mut result: HashSet<String> = HashSet::new();
+        let resource_list = &config.resources;
+        if resource_list.is_empty() {
+            return Vec::new();
+        }
+
+        for r in resource_list
+        {
+            let v = map.entry(&r.resource_type).or_insert(0);
+            *v += 1;
+            if *v > 1 {
+                result.insert(r.resource_type.clone());
+            }
+        }
+
+        result.into_iter().collect()
+    }
+
+    pub fn invoke_export(&self, _error_action: ErrorAction, _progress_callback: impl Fn() + 'static) -> Result<ConfigurationExportResult, DscError> {
+        let (config, messages, had_errors) = self.validate_config()?;
+
+        let duplicates = Self::find_duplicate_resource_types(&config);
+        if !duplicates.is_empty()
+        {
+            let duplicate = &duplicates[0];
```

Comment:
There can be multiple dupes here.  I think you can format this as an array:

`Resource(s) Name, Name2 specified multiple times`

There's probably a join function you can use to create the new string with commas.

Reply:
Updated.